pandas/core/indexes/base.py

```diff
@@ -4577,21 +4577,23 @@ def get_value(self, series, key):
         # use this, e.g. DatetimeIndex
         # Things like `Series._get_value` (via .at) pass the EA directly here.
         s = getattr(series, "_values", series)
-        if isinstance(s, (ExtensionArray, Index)) and is_scalar(key):
-            # GH 20882, 21257
-            # Unify Index and ExtensionArray treatment
-            # First try to convert the key to a location
-            # If that fails, raise a KeyError if an integer
-            # index, otherwise, see if key is an integer, and
-            # try that
-            try:
-                iloc = self.get_loc(key)
-                return s[iloc]
-            except KeyError:
-                if len(self) > 0 and (self.holds_integer() or self.is_boolean()):
-                    raise
-                elif is_integer(key):
-                    return s[key]
+        if isinstance(s, ExtensionArray):
```

Comment:
I removed the `Index` from this check, because the line above (`s = getattr(series, "_values", series)`) should nowadays extract the EA from the Index (and no Series is backed by an Index any more)

Reply:
maybe modernize to use extract_array?

Reply:
Indeed, used `extract_array`